src/instana/tracer.py

```diff
@@ -218,14 +213,17 @@ def _add_stack(self, span: InstanaSpan, limit: Optional[int] = 30) -> None:
     def _create_span_context(self, parent_context: SpanContext) -> SpanContext:
         """Creates a new SpanContext based on the given parent context."""
 
+        generated_id = generate_id()
+
         if parent_context is not None and parent_context.trace_id is not None:
             trace_id = parent_context.trace_id
-            span_id = generate_id()
+            span_id = generated_id
             trace_flags = parent_context.trace_flags
             is_remote = parent_context.is_remote
         else:
-            trace_id = self.tracer_id
-            span_id = self.tracer_id
+            # root span
+            trace_id = generated_id
+            span_id = generated_id
```

Comment:
I would make it a bit cleaner to read the code (and also a bit more efficient):

```suggestion
        if parent_context is not None and parent_context.trace_id is not None:
            trace_id = parent_context.trace_id
            span_id = generate_id()
            trace_flags = parent_context.trace_flags
            is_remote = parent_context.is_remote
        else:
            trace_id = span_id = generate_id()
```

Reply:
Done!